homeassistant/components/group/__init__.py

```diff
@@ -229,7 +232,9 @@ async def async_setup(hass, config):
 
     async def reload_service_handler(service: ServiceCall) -> None:
         """Remove all user-defined groups and load new ones from config."""
-        auto = list(filter(lambda e: not e.user_defined, component.entities))
+        auto: list[Group] = [
+            e for e in component.entities if isinstance(e, Group) and not e.user_defined
```

Comment:
Do we still need the isinstance check if we know that all entities of the component are Group instances? Maybe we just need to cast?